docs/user-guide/src/bmo/ironic_installation.md

```diff
@@ -0,0 +1,181 @@
+
+## Optional containers
+
+In some circumstances, it might be necessary to install some additional containers on
+top of Ironic containers. And this might depend on whether your target nodes are
+virtual or physical machines, or whether you want to use more stable BMC protocol
+like Redfish and etc. Let's check why we would need these optional containers.
+We run these optional containers on the host machine, i.e., out-of-cluster regardless
+of the Ironic installation method.
+
+### vbmc
+
+If you plan to manage [livbirt](https://en.wikipedia.org/wiki/Libvirt) Virtual Machine (VM),
+there is a high chance that you need to install Virtual Baseboard Management
+Controller (VBMC), which is virtual implementation of Baseboard Management
+Controller (BMC). For More info on VBMC check official [documentation](https://docs.openstack.org/virtualbmc/latest/).
+
+```bash
+ sudo docker run -d --net host \
+    --name vbmc \
+     -v "SOME_LOCAL_PATH":/root/.vbmc \
+     -v "/root/.ssh":/root/ssh \
+     quay.io/metal3-io/vbmc:latest
+```
+
+Our VMBC container image is available at [quay.io/metal3-io/vbmc](https://quay.io/repository/metal3-io/vbmc?tag=latest&tab=tags)
+
+### sushy-tools
+
+Sushy-tools is the same functionality as VBMC tool, except the fact it relies on [Redfish](https://www.dmtf.org/standards/redfish)
+protocol instead of IPMI. For more info, check official [documentation](https://docs.openstack.org/sushy-tools/latest/index.html).
+
+Susy-tool accepts some configuration parameters, which
+we pass in the form of a file. Here is an example of `conf.py`
+configuration file. To see full list of supported parameters,
+check sushy-tools [documentation](https://docs.openstack.org/sushy-tools/latest/admin/index.html).
+
+```python
+# The libvirt URI to use. This option enables libvirt drivers.
+# qemu+ssh://root@<remote-libvirt-server>/system?keyfile=<path-to-key>
+SUSHY_EMULATOR_LIBVIRT_URI = "qemu+ssh://root@192.168.111.1/system?&keyfile=/root/ssh/id_rsa_virt_power&no_verify=1&no_tty=1"
+SUSHY_EMULATOR_IGNORE_BOOT_DEVICE = False
+SUSHY_EMULATOR_VMEDIA_VERIFY_SSL = False
+```
+
+We will mount `conf.py` into the container's `/root/sushy/` path.
+
+```bash
+ sudo docker run -d --net host \
+    --name sushy-tools \
+     -v "SOME_PATH/conf.py":/root/sushy/conf.py \
+     -v "/root/.ssh":/root/ssh \
+     quay.io/metal3-io/sushy-tools:latest
+```
+
+Our sushy-tools container image is available at  [quay.io/metal3-io/sushy-tools](https://quay.io/repository/metal3-io/sushy-tools?tag=latest&tab=tags)
+
+### httpd
+
+We use this container to run Apache web server, which serves our target node images as well as Ironic
+Python Images (IPA) via http for PXE booting. We will mount locally available images container's
+`/shared` dir.
+
+We are not using actual Apache container image here, but instead [quay.io/metal3-io/ironic](https://quay.io/repository/metal3-io/ironic?tag=latest&tab=tags) which has `runhttpd` entry point to start off the httpd.
```

Comment:
We are actually using Apache, if the base image is httpd/apache.

Reply:
I can mention that it is based on Apache?

Reply:
That would be good.